crates/bevy_audio/src/sinks.rs

```diff
@@ -139,15 +213,42 @@ impl AudioSinkPlayback for AudioSink {
 #[derive(Component)]
 pub struct SpatialAudioSink {
     pub(crate) sink: SpatialSink,
+
+    /// Managed volume allows the sink to be muted without losing the user's
+    /// intended volume setting.
+    ///
+    /// This is used to restore the volume when [`unmute`](Self::unmute) is
+    /// called.
+    ///
+    /// If the sink is not muted, this is `None`.
+    ///
+    /// If the sink is muted, this is `Some(volume)` where `volume` is the
+    /// user's intended volume setting, even if the underlying sink's volume is
+    /// 0.
+    pub(crate) managed_volume: Option<f32>,
+}
+
+impl SpatialAudioSink {
+    /// Create a new spatial audio sink.
+    pub fn new(sink: SpatialSink) -> Self {
+        Self {
+            sink,
+            managed_volume: None,
+        }
+    }
 }
 
 impl AudioSinkPlayback for SpatialAudioSink {
     fn volume(&self) -> f32 {
         self.sink.volume()
```

Comment:
Did you forget to change it here?

Reply:
Good pickup thanks. This is now covered by the added unit test assuming CI passes it (creating a rodio `SpatialSink` in a test is different to creating a `Sink`, so I'm keen to see if CI can run it).

Reply:
Yeah CI doesn't like the spatial test, so removed it in https://github.com/bevyengine/bevy/pull/16813/commits/973cf24aee87386104b2662bbe9a9fe4651dac82